src/blackbox/reader.go

```diff
@@ -38,19 +38,19 @@ func NewFlightLogReader(opts FlightLogReaderOpts) *FlightLogReader {
 // Returns channel with successfully parsed frames and channel with errors.
 func (f *FlightLogReader) LoadFile(file io.Reader, ctx context.Context) (<-chan Frame, <-chan error) {
 	frameChan := make(chan Frame)
-	errChan := make(chan error)
+	errChan := make(chan error, 2)
```

Comment:
I was not sure if line #45 would be blocking for a non-buffered channel, so I made it buffered.
I'll later experiment if it blocks or not (or if you know for sure that it's non blocking, then please remove the buffer)

Reply:
Wouldn't it be even easier to add an additional argument for synchronous errors ? (even if it can appear confusing at a first sight to have two errors in the signature)

Currently we don't make any distinction between recoverable errors and fatal errors. For recoverable errors, I think it would be more readable to have a frame of type "broken" rather than a select to always read from those two channels. Also, this would preserved the ordering between frames and recoverable errors, which is not the case with `select` statement of multiple channels iirc.

If we go in that direction, the only error left would be the fatal one. At that point, it would probably not be worth having a channel just for them, and another frame of type "broken-stream" or with an attribute "not-recoverable" could help get rid of it. What do you think ?